ext/nokogiri/xml_reader.c

```diff
@@ -723,6 +723,8 @@ rb_xml_reader_encoding(VALUE rb_reader)
   constructor_encoding = rb_iv_get(rb_reader, "@encoding");
   if (RTEST(constructor_encoding)) {
     return constructor_encoding;
+  } else {
+      rb_iv_set(self, "@encoding", NOKOGIRI_STR_NEW2("UTF-8"));
```

Comment:
If you check the original change, this is not done inside this function
And the fix seems to be for a feature that is not present in this old version
```suggestion
```